src/dawn/CodeGen/CXXNaive/ASTStencilBody.h

```diff
@@ -0,0 +1,120 @@
+//===--------------------------------------------------------------------------------*- C++ -*-===//
+//                          _
+//                         | |
+//                       __| | __ ___      ___ ___
+//                      / _` |/ _` \ \ /\ / / '_  |
+//                     | (_| | (_| |\ V  V /| | | |
+//                      \__,_|\__,_| \_/\_/ |_| |_| - Compiler Toolchain
+//
+//
+//  This file is distributed under the MIT License (MIT).
+//  See LICENSE.txt for details.
+//
+//===------------------------------------------------------------------------------------------===//
+
+#pragma once
+
+#include "dawn/CodeGen/ASTCodeGenCXX.h"
+#include "dawn/Optimizer/Interval.h"
+#include "dawn/Support/StringUtil.h"
+#include <stack>
+#include <unordered_map>
+
+namespace dawn {
+
+class StencilInstantiation;
+class StencilFunctionInstantiation;
+
+namespace codegen {
+namespace cxxnaive {
+
+// @brief context of a stencil body
+// (pure stencil or a stencil function)
+enum class StencilContext { E_Stencil, E_StencilFunction };
```

Comment:
What is the Prefix E standing for? shouldn't it be SC_Stencil[Function]

Reply:
yes, I ll change it. E was for Enum. I was somehow missing a convention to distinguish what is an object, a pointer and a enum...